parsing/parser.mly

```diff
@@ -1684,8 +1684,6 @@ expr:
       { dotop_set ~loc:$sloc (Ldot($3, "." ^ $4 ^ "{}<-")) $1 $6 $9 }
   | expr attribute
       { Exp.attr $1 $2 }
-  | UNDERSCORE
-     { not_expecting $loc($1) "wildcard \"_\"" }
```

Comment:
Note: this was added by @alainfrisch in 110835872cd75a495c1a58937de5499815842397 to address [MPR#6400](https://caml.inria.fr/mantis/view.php?id=6400), namely the following usability issue with the previous grammar:

```
# [{pstr_desc = Pstr_extension (extension_of_error error, _)}];;
Error: Syntax error: operator expected.
```

Reply:
With your current patch applied, the new behavior is as follows:

```
# [{pstr_desc = Pstr_extension (extension_of_error error, _)}];;
                                                          ^
Error: Syntax error: ')' expected
Line 1, characters 29-30: This '(' might be unmatched
```